### ISSUE TITLE
Freeform multiwindow crash

### ISSUE THREAD

**Issue by shuvashish76** [celzero/rethink-app#391]
**Crash logs:**
```
FATAL EXCEPTION: main
Process: com.celzero.bravedns, PID: 25305
java.lang.IllegalStateException: exitFreeformMode: You can only go fullscreen from freeform.
	at android.os.Parcel.readException(Parcel.java:2021)
	at android.os.Parcel.readException(Parcel.java:1959)
	at android.app.IActivityManager$Stub$Proxy.exitFreeformMode(IActivityManager.java:9209)
	at android.app.Activity.exitFreeformMode(Activity.java:3237)
	at com.android.internal.widget.DecorCaptionView.maximizeWindow(DecorCaptionView.java:349)
	at com.android.internal.widget.DecorCaptionView.onSingleTapUp(DecorCaptionView.java:409)
	at android.view.GestureDetector.onTouchEvent(GestureDetector.java:640)
	at com.android.internal.widget.DecorCaptionView.onTouchEvent(DecorCaptionView.java:170)
	at android.view.View.dispatchTouchEvent(View.java:11777)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2962)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2643)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2657)
	at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:448)
	at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1829)
	at android.app.Activity.dispatchTouchEvent(Activity.java:3321)
	at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:69)
	at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:410)
	at android.view.View.dispatchPointerEvent(View.java:12016)
	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4809)
	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4623)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4161)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4214)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4180)
	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4307)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4188)
	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4364)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4161)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4214)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4180)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4188)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4161)
	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:6682)
	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:6656)
	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:6617)
	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:6785)
	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:187)
	at android.os.MessageQueue.nativePollOnce(Native Method)
	at android.os.MessageQueue.next(MessageQueue.java:325)
	at android.os.Looper.loop(Looper.java:142)
	at android.app.ActivityThread.main(ActivityThread.java:6518)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

**Device details:**
Model: Lenovo Tab4 8 plus
Android version: 8.1.0
App version: 053g (18) [F-Droid]

**Comment by ignoramous** [comment]
Thanks for the bug report. What are steps to reproduce this? Haven't got a tablet, would try these on an emulator.

**Comment by shuvashish76** [comment]
Another Freeform Multiwindow crash

**Crash logs:**
```
FATAL EXCEPTION: main
Process: com.celzero.bravedns, PID: 28828
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.celzero.bravedns/com.celzero.bravedns.ui.FirewallActivity}: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment com.celzero.bravedns.ui.ConnTrackerBottomSheetFragment: could not find Fragment constructor
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2781)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2859)
	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4702)
	at android.app.ActivityThread.-wrap18(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1598)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6518)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
Caused by: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment com.celzero.bravedns.ui.ConnTrackerBottomSheetFragment: could not find Fragment constructor
	at androidx.fragment.app.Fragment.instantiate(Fragment.java:628)
	at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57)
	at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:483)
	at androidx.fragment.app.FragmentStateManager.<init>(FragmentStateManager.java:85)
	at androidx.fragment.app.FragmentManager.restoreSaveState(FragmentManager.java:2728)
	at androidx.fragment.app.FragmentController.restoreSaveState(FragmentController.java:198)
	at androidx.fragment.app.FragmentActivity$2.onContextAvailable(FragmentActivity.java:149)
	at androidx.activity.contextaware.ContextAwareHelper.dispatchOnContextAvailable(ContextAwareHelper.java:99)
	at androidx.activity.ComponentActivity.onCreate(ComponentActivity.java:322)
	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:273)
	at com.celzero.bravedns.ui.FirewallActivity.onCreate(FirewallActivity.kt:55)
	at android.app.Activity.performCreate(Activity.java:7023)
	at android.app.Activity.performCreate(Activity.java:7014)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1215)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2734)
	... 10 more
Caused by: java.lang.NoSuchMethodException: <init> []
	at java.lang.Class.getConstructor0(Class.java:2320)
	at java.lang.Class.getConstructor(Class.java:1725)
	at androidx.fragment.app.Fragment.instantiate(Fragment.java:613)
	... 24 more
```

**Steps to produce:**

1. Firewall > Network Log > click on any app (_view bottom sheet_)
2. Now click on device "recent apps" > Freeform Multiwindow
3. App crashes

**Comment by ignoramous** [comment]
Fixed: #380

**Comment by ignoramous** [comment]
> What are steps to reproduce this? Haven't got a tablet, would try these on an emulator.

One way to invoke *freeform* (and hence the crashes) is to split screen a couple of rdns activity with another (unrelated app) activity and change the split screen's vertical height.

**Comment by ignoramous** [comment]
Fixed in `v053j` #525
